marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -19,24 +19,28 @@ const ds = require("/data-hub/5/data-services/ds-utils.sjs");
 // TODO Will move this to /data-hub/5/entities soon
 const entityLib = require("/data-hub/5/impl/entity-lib.sjs");
 
+let selectedPropertyMetadata = null;
```

Comment:
Let's make these parameters that are passed to functions instead of them being library-level variables. That makes the scope of functions easier to understand, as they won't be depending on library-level variables where it's not clear when the state of those variables is modified.